### DISCUSSION TITLE
Github workflow is free for public repos?

### DISCUSSION THREAD

**Discussion by lakshayrohila** [puppylinux-woof-CE/woof-CE#3579] [Q&A] [answered]
According to [this](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions), github workflows are free for public repos, i.e. no limitation on minutes ane storage space for public repos. But according to [this](https://github.com/puppylinux-woof-CE/woof-CE/wiki/Building-a-Puppy-on-GitHub), github repos are not free to use github workflows without any restrictions. Which one is correct?

**Comment by dimkr** [comment, marked as answer]
It's free, but with usage limits.

**Reply by lakshayrohila** [reply]
What's the usage limit then (I need this because I myself use Github Actions for other programs I have developed)?

**Reply by dimkr** [reply]
What GitHub's documentation says, https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration

**Reply by lakshayrohila** [reply]
But I find the same text in https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#about-billing-for-github-actions (the same page you referred). I mean if that were true, and total space we'd get is **500MBs** free, then most of the puppy builds will not be successful. Many puppy builds cross that limit already.

**Reply by dimkr** [reply]
500 MB relates to packages, not actions artifacts and releases

**Reply by dimkr** [reply]
(By the way, I don't know where to find that in the documentation, but traffic is limited too. The woof-CE CI builds easily exhaust the limit and downloads or uploads start to fail.)

**Reply by lakshayrohila** [reply]
Thanks! I understand now.
